Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -31,8 +31,92 @@
 #import <algorithm>
 #import <vector>
 
+#import <CoreFoundation/CoreFoundation.h>
+#import <CFRuntime.h>
+#import "D2DWrapper.h"
+
+#include <COMIncludes.h>
+#import <wrl/client.h>
+#import <D2d1.h>
+#include <COMIncludes_End.h>
+
 static const wchar_t* TAG = L"CGPath";
 
+using namespace std;
+using namespace Microsoft::WRL;
+
+struct __CGPath {
+    CFRuntimeBase _base;
+
+    ComPtr<ID2D1PathGeometry> _pathGeometry;
+    ComPtr<ID2D1GeometrySink> _geometrySink;
+
+    bool _isPathClosed;
+    CGPoint _currentPoint;
+    CGPoint _startingPoint;
+};
+
+// A private helper function for re-opening a path geometry.
+static void __preparePathForEditing(CGPathRef path) {
```

Comment:
>__preparePathForEditing [](start = 12, length = 23)

Nit: these could be member functions of __CGPath.  A struct is a class afterall.